src/operation/update.rs

```diff
@@ -144,18 +146,15 @@ impl OperationWithDefaults for Update {
             }
 
             if let Some(ref let_vars) = options.let_vars {
-                body.append(
-                    cstr!("let"),
-                    crate::bson_compat::serialize_to_raw_document_buf(&let_vars)?,
-                );
+                body.append(cstr!("let"), RawDocumentBuf::try_from(let_vars)?);
```

Comment:
Also unrelated but a nice example of why it's valuable to have serialization be visually distinct from encoding - happened to notice that this didn't need to go via serde as I was reading through.